src/Enum.php

```diff
@@ -224,8 +224,23 @@ public static function isValid($value)
      * Asserts valid enum value
      *
      * @psalm-pure
-     * @psalm-assert T $value
+     * @psalm-template CheckedValueType
+     * @psalm-param class-string<self<CheckedValueType>> $enumType
      * @param mixed $value
+     * @psalm-assert-if-true CheckedValueType $value
+     */
+    public static function isValidEnumValue(string $enumType, $value): bool
+    {
+        return $enumType::isValid($value);
+    }
+
+    /**
+     * Asserts valid enum value
+     *
+     * @psalm-pure
+     * @param mixed $value
+     *
+     * deprecated use {@see Enum::assertValidEnumValue()} instead
```

Comment:
This is not yet `@deprecated` - should I mark it as such, or should I drop the comment and leave it be?